src/Paket.Bootstrapper/Program.cs

```diff
@@ -23,9 +23,13 @@ static void Main(string[] args)
             Console.CancelKeyPress += CancelKeyPressed;
 
             var fileProxy = new FileSystemProxy();
+            var optionsBeforeDependenciesFile = ArgumentParser.ParseArgumentsAndConfigurations(args, ConfigurationManager.AppSettings,
+                Environment.GetEnvironmentVariables(), fileProxy, Enumerable.Empty<string>());
+            ConsoleImpl.Verbosity = optionsBeforeDependenciesFile.Verbosity;
```

Comment:
Maybe the check for the environment variable should be done directly at that level without calling the ArgumentParser.

Accepting that the only way to verbose-log the early/configuration part of the bootstrapper is the environment variable.